src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -159,17 +170,52 @@ public IReadOnlyList<SupportedMode> KnownModes
         public static bool TryConnectToNetworkedBoard(IPAddress boardAddress, int port,
 #if NET5_0_OR_GREATER
             [NotNullWhen(true)]
+#endif
+            out ArduinoBoard? board)
+        {
+            return TryConnectToNetworkedBoard(boardAddress, port, true, out board);
+        }
+
+        /// <summary>
+        /// Tries to connect to an arduino over network.
+        /// This requires an arduino with an ethernet shield or an ESP32 with enabled WIFI support.
+        /// </summary>
+        /// <param name="boardAddress">The IP address of the board</param>
+        /// <param name="port">The network port to use. The default port is 27016</param>
+        /// <param name="useAutoReconnect">True to use an auto-reconnecting stream. Helpful when using an unreliable connection.</param>
+        /// <param name="board">Returns the board if successful</param>
+        /// <returns>True on success, false otherwise</returns>
+        public static bool TryConnectToNetworkedBoard(IPAddress boardAddress, int port, bool useAutoReconnect,
+#if NET5_0_OR_GREATER
+            [NotNullWhen(true)]
 #endif
             out ArduinoBoard? board)
         {
             board = null;
```

Comment:
Nit: it is usually better to set the out parameter right next to the return statement so that assignment is clearer when reading the code.